distributed/diagnostics/nvml.py

```diff
@@ -63,6 +65,34 @@ def _pynvml_handles():
     return pynvml.nvmlDeviceGetHandleByIndex(gpu_idx)
 
 
+def _pynvml_not_supported():
+    h = _pynvml_handles()
+
+    try:
+        util = pynvml.nvmlDeviceGetUtilizationRates(h).gpu
+    except pynvml.NVMLError_NotSupported:
+        util = None
+    try:
+        mem = pynvml.nvmlDeviceGetMemoryInfo(h).used
+    except pynvml.NVMLError_NotSupported:
+        mem = None
+    try:
+        total = pynvml.nvmlDeviceGetMemoryInfo(h).total
+    except pynvml.NVMLError_NotSupported:
+        total = None
+    try:
+        name = pynvml.nvmlDeviceGetName(h).decode()
+    except pynvml.NVMLError_NotSupported:
+        name = None
+
+    return {
+        "util": util is None,
+        "mem": mem is None,
+        "total": total is None,
+        "name": name is None,
+    }
```

Comment:
The way this is written/being used seems overly complex to me. Any reason why we couldn't define a function for each of those attributes returning either the result or `None` depending on it being supported, therefore avoiding the extra check such as the one in https://github.com/dask/distributed/pull/5343/files#diff-457ccaa4d4a70945b095c0460229791e8142c269e9ec781b98ab6d1b1925fd8cR117-R119 ?

Reply:
Were you thinking something like:

```python
def _nvml_util():
    global nvmlUtilSupported
    if nvmlUtilSupported:
        return pynvml.nvmlDeviceGetUtilizationRates(h).gpu
    try:
        util = pynvml.nvmlDeviceGetUtilizationRates(h).gpu
        nvmlUtilSupported = True
        return util
    except pynvml.NVMLError_NotSupported:
        return None
```

Reply:
Yes, that. I'm still wondering if the global is at all necessary. Do you see any considerable difference in performance if the entire function is the `try..except` block?

Reply:
It doesn't seem like performance is impacted; in that case, I'm generally okay with having the check be just the `try..except` block.